src/Analyzers/CSharp/CodeFixes/UseParameterNullChecking/CSharpUseParameterNullCheckingCodeFixProvider.cs

```diff
@@ -0,0 +1,84 @@
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.UseParameterNullChecking
+{
+    [ExportCodeFixProvider(LanguageNames.CSharp, Name = PredefinedCodeFixProviderNames.UseIsNullCheck), Shared]
+    internal class CSharpUseParameterNullCheckingCodeFixProvider : SyntaxEditorBasedCodeFixProvider
+    {
+        [ImportingConstructor]
+        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+        public CSharpUseParameterNullCheckingCodeFixProvider()
+        {
+        }
+
+        public override ImmutableArray<string> FixableDiagnosticIds
+            => ImmutableArray.Create(IDEDiagnosticIds.UseParameterNullCheckingId);
+
+        internal sealed override CodeFixCategory CodeFixCategory => CodeFixCategory.CodeStyle;
+
+        public override Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            var diagnostic = context.Diagnostics[0];
+            context.RegisterCodeFix(
+                new MyCodeAction(CSharpAnalyzersResources.Use_parameter_null_checking,
+                c => FixAsync(context.Document, diagnostic, c)),
+                context.Diagnostics);
+
+            return Task.CompletedTask;
+        }
+
+        protected override Task FixAllAsync(
+            Document document, ImmutableArray<Diagnostic> diagnostics,
+            SyntaxEditor editor, CancellationToken cancellationToken)
+        {
+            foreach (var diagnostic in diagnostics)
+            {
+                var node = diagnostic.Location.FindNode(getInnermostNodeForTie: true, cancellationToken: cancellationToken);
+                switch (node)
+                {
+                    case IfStatementSyntax ifStatement:
+                        editor.RemoveNode(ifStatement);
+                        break;
+                    case ExpressionStatementSyntax expressionStatement:
+                        // this.item = item ?? throw new ArgumentNullException(nameof(item));
+                        var assignment = (AssignmentExpressionSyntax)expressionStatement.Expression;
+                        var nullCoalescing = (BinaryExpressionSyntax)assignment.Right;
+                        var parameterReferenceSyntax = nullCoalescing.Left;
+                        editor.ReplaceNode(nullCoalescing, parameterReferenceSyntax.WithAppendedTrailingTrivia(SyntaxFactory.ElasticMarker));
+                        break;
+                    default:
+                        throw ExceptionUtilities.UnexpectedValue(node);
+                }
+
+                var parameterSyntax = (ParameterSyntax)diagnostic.AdditionalLocations[0].FindNode(cancellationToken);
```

Comment:
if you're going to presuem this is a ParameterSyntax, also do a check that it's that type in the analyzer.  note: i might recommend using IParameterSymbol.DeclaringSyntaxReferences (instead of Locations).  The former tends to correspond directly to the syntax node (where the latter often refers to token names and can be iffy on recovery).

Reply:
have added the check

https://github.com/dotnet/roslyn/blob/bdda177645ef4df04ab8044a53738708ac771cd4/src/Analyzers/CSharp/Analyzers/UseParameterNullChecking/CSharpUseParameterNullCheckingDiagnosticAnalyzer.cs#L187-L197